File: src/Entity/Credit.php
<?php

namespace App\Entity;

use App\Entity\Echeance;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\CreditRepository;
use Doctrine\Common\Collections\Collection;
use Doctrine\Common\Collections\ArrayCollection;

#[ORM\Entity(repositoryClass: CreditRepository::class)]
class Credit
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(inversedBy: 'credits')]
    private ?Compte $id_compte = null;

    #[ORM\Column]
    private ?float $montant = null;

    #[ORM\Column]
    private ?float $taux = null;

    #[ORM\Column]
    private ?int $duree = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $date_debut = null;

    #[ORM\Column(length: 255)]
    private ?string $type = null;

    #[ORM\Column]
    private ?float $apport_propre = null;

    #[ORM\Column]
    private ?float $revenu_propre = null;

    #[ORM\Column]
    private ?float $montant_echeance = null;

    #[ORM\Column]
    private ?int $date_echeance = null;

    #[ORM\Column(length: 255)]
    private ?string $etat = null;

    #[ORM\Column]
    private ?int $nb_echeances_restants = null;

    #[ORM\OneToMany(mappedBy: 'id_credit', targetEntity: Echeance::class)]
    private Collection $echeances;


    public function __construct()
    {
        $this->echeances = new ArrayCollection();
      
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getIdCompte(): ?Compte
    {
        return $this->id_compte;
    }

    public function setIdCompte(?Compte $id_compte): static
    {
        $this->id_compte = $id_compte;

        return $this;
    }

    public function getMontant(): ?float
    {
        return $this->montant;
    }

    public function setMontant(float $montant): static
    {
        $this->montant = $montant;

        return $this;
    }

    public function getTaux(): ?float
    {
        return $this->taux;
    }

    public function setTaux(float $taux): static
    {
        $this->taux = $taux;

        return $this;
    }

    public function getDuree(): ?int
    {
        return $this->duree;
    }

    public function setDuree(int $duree): static
    {
        $this->duree = $duree;

        return $this;
    }

    public function getDateDebut(): ?\DateTimeInterface
    {
        return $this->date_debut;
    }

    public function setDateDebut(\DateTimeInterface $date_debut): static
    {
        $this->date_debut = $date_debut;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;

        return $this;
    }

    public function getApportPropre(): ?float
    {
        return $this->apport_propre;
    }

    public function setApportPropre(float $apport_propre): static
    {
        $this->apport_propre = $apport_propre;

        return $this;
    }

    public function getRevenuPropre(): ?float
    {
        return $this->revenu_propre;
    }

    public function setRevenuPropre(float $revenu_propre): static
    {
        $this->revenu_propre = $revenu_propre;

        return $this;
    }

    public function getMontantEcheance(): ?float
    {
        return $this->montant_echeance;
    }

    public function setMontantEcheance(float $montant_echeance): static
    {
        $this->montant_echeance = $montant_echeance;

        return $this;
    }

    public function getDateEcheance(): ?int
    {
        return $this->date_echeance;
    }

    public function setDateEcheance(int $date_echeance): static
    {
        $this->date_echeance = $date_echeance;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): static
    {
        $this->etat = $etat;

        return $this;
    }

    public function getNbEcheancesRestants(): ?int
    {
        return $this->nb_echeances_restants;
    }

    public function setNbEcheancesRestants(int $nb_echeances_restants): static
    {
        $this->nb_echeances_restants = $nb_echeances_restants;

        return $this;
    }

    /**
     * @return Collection<int, Echeance>
     */
    public function getEcheances(): Collection
    {
        return $this->echeances;
    }

    public function addEcheance(Echeance $echeance): static
    {
        if (!$this->echeances->contains($echeance)) {
            $this->echeances->add($echeance);
            $echeance->setIdCredit($this);
        }

        return $this;
    }


    public function setEcheances(ArrayCollection $echeances): static
    {
       $this->echeances=$echeances;

        return $this;
    }

    public function removeEcheance(Echeance $echeance): static
    {
        if ($this->echeances->removeElement($echeance)) {
            // set the owning side to null (unless already changed)
            if ($echeance->getIdCredit() === $this) {
                $echeance->setIdCredit(null);
            }
        }

        return $this;
    }
}

File: src/main/java/com/inventorymanagementsystem/entity/Credit.java
package com.inventorymanagementsystem.entity;


import java.sql.Date;
public class Credit implements Comparable<Credit> {

    private Integer id;
    private Integer id_compte_id;
    private Double montant;
    private Double taux;
    private Integer duree;
    private Date date_debut;
    private String type;
    private Double apport_propre;
    private Double revenu_propre;
    private Double montant_echeance;
    private Integer date_echeance;
    private String etat;
    private Integer nb_echeances_restants;

    public Credit(Integer id_compte_id, Double montant, Double taux, Integer duree, Date date_debut, String type, Double apport_propre, Double revenu_propre, Double montant_echeance, Integer date_echeance, String etat, Integer nb_echeances_restants) {
        this.id_compte_id = id_compte_id;
        this.montant = montant;
        this.taux = taux;
        this.duree = duree;
        this.date_debut = date_debut;
        this.type = type;
        this.apport_propre = apport_propre;
        this.revenu_propre = revenu_propre;
        this.montant_echeance = montant_echeance;
        this.date_echeance = date_echeance;
        this.etat = etat;
        this.nb_echeances_restants = nb_echeances_restants;
    }

    public Credit(Integer id, Integer id_compte_id, Double montant, Double taux, Integer duree, Date date_debut, String type, Double apport_propre, Double revenu_propre, Double montant_echeance, Integer date_echeance, String etat, Integer nb_echeances_restants) {
        this.id = id;
        this.id_compte_id = id_compte_id;
        this.montant = montant;
        this.taux = taux;
        this.duree = duree;
        this.date_debut = date_debut;
        this.type = type;
        this.apport_propre = apport_propre;
        this.revenu_propre = revenu_propre;
        this.montant_echeance = montant_echeance;
        this.date_echeance = date_echeance;
        this.etat = etat;
        this.nb_echeances_restants = nb_echeances_restants;
    }

    @Override
    public String toString() {
        return "Credit{" +
                "id=" + id +
                ", id_compte_id=" + id_compte_id +
                ", montant=" + montant +
                ", taux=" + taux +
                ", duree=" + duree +
                ", date_debut=" + date_debut +
                ", type='" + type + '\'' +
                ", apport_propre=" + apport_propre +
                ", revenu_propre=" + revenu_propre +
                ", montant_echeance=" + montant_echeance +
                ", date_echeance=" + date_echeance +
                ", etat='" + etat + '\'' +
                ", nb_echeances_restants=" + nb_echeances_restants +
                '}';
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public void setId_compte_id(Integer id_compte_id) {
        this.id_compte_id = id_compte_id;
    }

    public void setMontant(Double montant) {
        this.montant = montant;
    }

    public void setTaux(Double taux) {
        this.taux = taux;
    }

    public void setDuree(Integer duree) {
        this.duree = duree;
    }

    public void setDate_debut(Date date_debut) {
        this.date_debut = date_debut;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setApport_propre(Double apport_propre) {
        this.apport_propre = apport_propre;
    }

    public void setrevenu_propre(Double revenu_propre) {
        this.revenu_propre = revenu_propre;
    }

    public void setmontant_echeance(Double montant_echeance) {
        this.montant_echeance = montant_echeance;
    }

    public void setDate_echeance(Integer date_echeance) {
        this.date_echeance = date_echeance;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public void setNb_echeances_restants(Integer nb_echeances_restants) {
        this.nb_echeances_restants = nb_echeances_restants;
    }

    public Integer getId() {
        return id;
    }

    public Integer getId_compte_id() {
        return id_compte_id;
    }

    public Double getMontant() {
        return montant;
    }

    public Double getTaux() {
        return taux;
    }

    public Integer getDuree() {
        return duree;
    }

    public Date getDate_debut() {
        return date_debut;
    }

    public String getType() {
        return type;
    }

    public Double getApport_propre() {
        return apport_propre;
    }

    public Double getrevenu_propre() {
        return revenu_propre;
    }

    public Double getMontant_echeance() {
        return montant_echeance;
    }

    public Integer getDate_echeance() {
        return date_echeance;
    }

    public String getEtat() {
        return etat;
    }

    public Integer getNb_echeances_restants() {
        return nb_echeances_restants;
    }

    @Override
    public int compareTo( Credit credit) {
        return 0;
    }



}